Objects/genericaliasobject.c

```diff
@@ -285,45 +336,15 @@ ga_getitem(PyObject *self, PyObject *item)
         if (typevar) {
             Py_ssize_t iparam = tuple_index(alias->parameters, nparams, arg);
             assert(iparam >= 0);
-            arg = args[iparam];
+            arg = argitems[iparam];
             Py_INCREF(arg);
         }
         else {
-            _Py_IDENTIFIER(__parameters__);
-            PyObject *subparams;
-            if (_PyObject_LookupAttrId(arg, &PyId___parameters__, &subparams) < 0) {
+            arg = subs_tvars(arg, alias->parameters, argitems);
+            if (arg == NULL) {
                 Py_DECREF(newargs);
                 return NULL;
             }
-            if (subparams && PyTuple_Check(subparams)) {
-                Py_ssize_t nsubargs = PyTuple_GET_SIZE(subparams);
-                PyObject *newsubargs = PyTuple_New(nsubargs);
-                if (newsubargs == NULL) {
-                    Py_DECREF(subparams);
-                    Py_DECREF(newargs);
-                    return NULL;
-                }
-                for (Py_ssize_t j = 0; j < nsubargs; ++j) {
-                    PyObject *a = PyTuple_GET_ITEM(subparams, j);
-                    Py_ssize_t iparam = tuple_index(alias->parameters, nparams, a);
-                    if (iparam >= 0) {
-                        a = args[iparam];
-                    }
-                    Py_INCREF(a);
-                    PyTuple_SET_ITEM(newsubargs, j, a);
-                }
-                arg = PyObject_GetItem(arg, newsubargs);
-                Py_DECREF(newsubargs);
-                if (arg == NULL) {
-                    Py_DECREF(subparams);
-                    Py_DECREF(newargs);
-                    return NULL;
-                }
-            }
-            else {
-                Py_INCREF(arg);
-            }
-            Py_XDECREF(subparams);
         }
         PyTuple_SET_ITEM(newargs, iarg, arg);
     }
```

Comment:
Comment for the next line: (again) since it is the main event we've been working up to, offset it with a blank line before and after.